### PR TITLE
Fix the bug that the emoji does not disappear

### LINKED ISSUE #11036: Reverting a reaction gives no feedback
- Gitea version (or commit ref): 1.12.0+dev-127-gd21b974f6
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes (provide example URL): https://try.gitea.io/Septatrix/HelloTea/issues/3
  - [ ] No
  - [ ] Not relevant
- Browser: Chromium 80.0.3987.163

## Description
When adding an emoji reaction to an issue it gets added instantaneously. Reverting said reaction however yields no visible feedback. After clicking on the icon nothing happens until one refreshes the website.
I only noticed that it had an effect because I was double checking before creating an issue about there being no way to retract a reaction.

### PR DESCRIPTION
**Pull request by L0veSunshine**
Fixes #11036

When you add **one** emoji here. and then cancel this emoji. The emoji will not disapper until you fresh the page.(This only occur when there is only one emoji)
This bug is on https://try.gitea.io/ and https://gitea.com/. You can have a try. **Click here!** 
image

### CODE REVIEW

**Comment by CirnoT** [comment]
Can you explain better what exactly does this fixes?

**Comment by L0veSunshine** [comment]
> Can you explain better what exactly does this fixes?

When there is no emojy the ajax XHR is like this:
image

**Comment by CirnoT** [comment]
For anyone wondering, the issue is that attempting to unreact when there's only one emoji left has no effect on UI

**Comment by L0veSunshine** [comment]
> For anyone wondering, the issue is that attempting to unreact when there's only one emoji left has no effect on UI

Yes, that's what it means

File: web_src/js/index.js [hunk 1]
```diff
@@ -242,6 +242,8 @@ function initReactionSelector(parent) {
         let react = content.find('.segment.reactions');
         if (!resp.empty && react.length > 0) {
           react.remove();
+        } else if (resp.empty && resp.html === '') {
+          react.remove();
```
Comment (CirnoT) [inline web_src/js/index.js:243]: Shouldn't we still ensure that `react.length` is non-zero here?

Comment (L0veSunshine) [reply web_src/js/index.js:243]: When click to unreact emoji and there has no emoji left. The ajax XHR looks like this:
image

Comment (L0veSunshine) [reply web_src/js/index.js:243]: This XHR response is unique. Don't need to do the superfluous check.

Comment (CirnoT) [reply web_src/js/index.js:243]: `react.length` has nothing to do with `resp` however? You don't want to do `.remove` on something that doesn't exist.

Comment (L0veSunshine) [reply web_src/js/index.js:243]: "react" is the jquery selector. "resp" is the XMLHTTPRequest's alias.
Build a new version with this js file and have a try.

Comment (CirnoT) [reply web_src/js/index.js:243]: What I am concerned about is executing `.remove` on `react` if user clicks two times.

File: web_src/js/index.js [hunk 2]
```diff
@@ -242,6 +242,8 @@ function initReactionSelector(parent) {
         let react = content.find('.segment.reactions');
         if (!resp.empty && react.length > 0) {
```
Comment (lafriks) [inline web_src/js/index.js]: This change should do the same, this way no new else if is needed
```suggestion
        if ((!resp.empty || resp.html === '') && react.length > 0) {
```

File: web_src/js/index.js [hunk 3]
```diff
@@ -240,7 +240,9 @@ function initReactionSelector(parent) {
       if (resp && (resp.html || resp.empty)) {
         const content = $(vm).closest('.content');
         let react = content.find('.segment.reactions');
-        if (!resp.empty && react.length > 0) {
+        if ((!resp.empty || resp.html === '') && react.length > 0) {
+          react.remove();
+        } else if (resp.empty && resp.html === '') {
```
Comment (lafriks) [inline web_src/js/index.js]:
```suggestion
```

Comment (lafriks) [reply web_src/js/index.js]: not needed anymore

**Review by CirnoT — APPROVED** [review]

**Review by CirnoT — APPROVED** [review]

**Review by silverwind — APPROVED** [review]

**Review by lafriks — APPROVED** [review]

**Comment by techknowlogick** [comment]
ping LG-TM

**Comment by techknowlogick** [comment]
@L0veSunshine please send backport to release/v1.12 :)

**Pull request merged**